test/Table_Tests/src/Common_Table_Operations/Map_Spec.enso

```diff
@@ -39,13 +43,94 @@ spec setup =
+                c2_1.value_type . should_equal Value_Type.Byte
+                c2_1.to_vector . should_equal [3, 5, 7]
+
+                c2_2 = t.at "X" . map f expected_value_type=(Value_Type.Integer Bits.Bits_16)
+                c2_2.value_type . should_equal (Value_Type.Integer Bits.Bits_16)
+                c2_2.to_vector . should_equal [3, 5, 7]
+
+                c3 = t.at "X" . map f expected_value_type=Value_Type.Mixed
+                c3.value_type . should_equal Value_Type.Mixed
+                c3.to_vector . should_equal [3, 5, 7]
+
+                g x = "|" + x.to_text + "|"
+                c4 = t.at "X" . map g
+                c4.value_type . should_equal Value_Type.Char
+                c4.to_vector . should_equal ["|1|", "|2|", "|3|"]
+
+                c5 = t.at "X" . map g expected_value_type=(Value_Type.Char variable_length=False size=3)
+                c5.value_type . should_equal (Value_Type.Char variable_length=False size=3)
+                c5.to_vector . should_equal ["|1|", "|2|", "|3|"]
+
+                h x =
+                    if x == 2 then "A" else x+1
+                c6 = t.at "X" . map h expected_value_type=Value_Type.Mixed
+                c6.value_type . should_equal Value_Type.Mixed
+                c6.to_vector . should_equal [2, "A", 4]
+
+                c7 = t.at "X" . map h
+                c7.value_type . should_equal Value_Type.Mixed
+                c7.to_vector . should_equal [2, "A", 4]
+
+            Test.specify "should fail with Invalid_Value_Type if some results do not match the expected_value_type" <|
+                t = table_builder [["X", [1, 2, 3]]]
+                c1 = t.at "X"
+
+                f x = 127^x
+                r1 = c1.map f expected_value_type=Value_Type.Byte
+                r1.should_fail_with Invalid_Value_Type
+                r1.catch.to_display_text . should_contain "Expected type Byte, but got a value 16129 of type Integer (16 bits)"
+
+                r2 = c1.map f expected_value_type=(Value_Type.Integer Bits.Bits_16)
+                r2.should_fail_with Invalid_Value_Type
+                r2.catch.to_display_text . should_contain "Expected type Integer (16 bits), but got a value 2048383 of type Integer (32 bits)"
+
+                g x = "|" + (x * 4).to_text + "|"
+                r3 = c1.map g expected_value_type=(Value_Type.Char variable_length=True size=3)
+                r3.should_fail_with Invalid_Value_Type
+                r3.catch.to_display_text . should_contain "Expected type Char (variable length, max_size=3), but got a value |12| of type Char (fixed length, size=4)"
+
+                h x = "_"+x.to_text
+                r4 = c1.map h expected_value_type=(Value_Type.Char variable_length=False size=3)
+                r4.should_fail_with Invalid_Value_Type
+                r4.catch.to_display_text . should_contain "Expected type Char (fixed length, size=3), but got a value _1 of type Char (fixed length, size=2)"
```

Comment:
Curious -- why is .catch needed here?

Reply:
It would likely work without it, but on principle I want to `catch` the dataflow error and lift it to value and inspect its display text, not the display text of the 'error value'. It's subtle and in case it is `should_contain` does not really matter. It would matter if I used `should_equal` - without `catch` I'd then get a `Error: ` prefix.